src/jit/lsra.cpp

```diff
@@ -2135,6 +2193,24 @@ void LinearScan::identifyCandidates()
         }
     }
 
+#if DOUBLE_ALIGN
+    if (checkDoubleAlign)
+    {
+        // TODO-CQ: Fine-tune this:
+        // In the legacy reg predictor, this runs after allocation, and then demotes any lclVars
+        // allocated to the frame pointer, which is probably the wrong order.
+        // However, because it runs after allocation, it can determine the impact of demoting
+        // the lclVars allocated to the frame pointer.
+        // => Here, estimate of the EBP refCnt and weighted refCnt is a wild guess.
+        //
+        unsigned refCntEBP    = refCntReg / 8;
+        unsigned refCntWtdEBP = refCntWtdReg / 8;
```

Comment:
You should call isFramePointerRequired() here and set these two values to zero if it returns true.
We can only use EBP for LclVar when this method returns false.